File: AR7ExportToolbar/ToolStripExport/Exporters/TextSectionReportExporter.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Text;
using GrapeCity.ActiveReports.Document;
using GrapeCity.ActiveReports.Export;

namespace AR7ExportToolbar.ToolStripExport.Exporters
{
	public class TextSectionReportExporter : SectionReportExporter
	{
		private const string TypeName = "GrapeCity.ActiveReports.Export.Xml.Section.TextExport";
		private const string AssemblyName = "GrapeCity.ActiveReports.Export.Xml.v7";

		public TextSectionReportExporter()
			: base(TypeName, AssemblyName)
		{
		}

		public override void Export(SectionDocument document, string filename, NameValueCollection settings)
		{
			throw new NotImplementedException();
		}

		public override string DefaultExtension
		{
			get { return "txt"; }
		}

		public override string FileDialogFilter
		{
			get { return "Text Files (*.txt)|*.txt|All Files (*.*)|*.*"; }
		}
	}
}

File: AR7ExportToolbarVB/ToolStripExport/Exporters/TextSectionReportExporter.vb
Imports GrapeCity.ActiveReports.Document
Imports System.Collections.Specialized

Public Class TextSectionReportExporter
	Inherits SectionReportExporter
	Private Const TypeName As String = "GrapeCity.ActiveReports.Export.Xml.Section.TextExport"
	Private Const AssemblyName As String = "GrapeCity.ActiveReports.Export.Xml.v7"

	Public Sub New()
		MyBase.New(TypeName, AssemblyName)
	End Sub

	Public Overrides Sub Export(document As SectionDocument, filename As String, settings As NameValueCollection)
		Throw New NotImplementedException()
	End Sub

	Public Overrides ReadOnly Property DefaultExtension() As String
		Get
			Return "txt"
		End Get
	End Property

	Public Overrides ReadOnly Property FileDialogFilter() As String
		Get
			Return "Text Files (*.txt)|*.txt|All Files (*.*)|*.*"
		End Get
	End Property
End Class